src/utils.ts

```diff
@@ -251,12 +251,29 @@ export function decorateWithReadConcern(
  * @param command - the command on which to apply the explain
  * @param options - the options containing the explain verbosity
  */
-export function decorateWithExplain(command: Document, explain: Explain): Document {
-  if (command.explain) {
+export function decorateWithExplain(
+  command: Document,
+  explain: Explain
+): {
+  explain: Document;
+  verbosity: ExplainVerbosity;
+  maxTimeMS?: number;
+} {
+  type ExplainCommand = ReturnType<typeof decorateWithExplain>;
+  const isExplainCommand = (doc: Document): doc is ExplainCommand => 'explain' in command;
```

Comment:
Why not define these outside of this function?

Reply:
They're an implementation detail and aren't used anywhere else, so they shouldn't be available anywhere else.

Reply:
Why is the condition now explain "in" instead of explain truthy?

Reply:
I don't care about the value, only that the key is present.

Reply:
What changed from before that we no longer care about the truthiness of the value? Doesn't an explain always have to be a document so it needs to be a non-null object or Map? It appears as though this has a functional change

Reply:
Sure, it's a change in behavior.  If you see my response to your comment above though, I've determined that it's safe to make.  We never decorate explain commands twice, we could alternatively just remove this logic.

Reply:
But also, not validating the type of this field aligns with our philosophy of not validating user input and instead letting the server error.

Reply:
@nbbeeken I left the type in, only because its useful when typing the `command` being built.  Since I initialize it to a literal, TS infers the type as { explain: Document, verbosity: string }.  If I don't type it like this (or as any or Document or something), TS complains that `maxTimeMS` isn't allowed on the command when I attach it.